tools/cldr-code/src/main/java/org/unicode/cldr/test/RelatedDatePathValues.java

```diff
@@ -0,0 +1,226 @@
+package org.unicode.cldr.test;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.ImmutableSet;
+import com.ibm.icu.text.UnicodeSet;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutionException;
+import org.unicode.cldr.util.CLDRFile;
+import org.unicode.cldr.util.CldrIntervalFormat;
+import org.unicode.cldr.util.XPathParts;
+
+public class RelatedDatePathValues {
+    public static final int calendarElement = 3;
+    public static final int dateTypeElement = 5;
+    public static final int idElement = 6;
```

Comment:
A comment like this would help:

In a path like `//ldml/dates/calendars/calendar[@type=\"gregorian\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\"MMMd\"]`,

element 0 = ldml
element 3 = calendarElement is like `calendar[@type=\"gregorian\"]`
element 5 = dateTypeElement is like `availableFormats`
element 6 = idElement is like `dateFormatItem[@id=\"MMMd\"]`

Reply:
could actually do it this way:

```js
XPathParts x = XPathParts.cloneAsThawed("//ldml/dates/calendars/calendar[@type=\"gregorian\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\"MMMd\"]);
public static final int calendarElement = x.findElement(LDMLConstants.CALENDAR);
public static final int dateTypeElement = x.findElement(LDMLConstants.DATE_TIME_FORMATS);
…
```

Reply:
I looked at this. For now I'm going to add comments. BTW LDMLConstants is itself kinda clunky, and requires manual update. Better would be to build constants (even better, an enum) from DtdData (but not now).